client/src/main/scala/skuber/api/package.scala

```diff
@@ -688,13 +688,17 @@ package object client {
             val path = Paths.get(new URL(fileUrl).toURI)
             Configuration.parseKubeconfigFile(path).get
           case None =>
-            // try KUBECONFIG - if that is not set then use default kubeconfig location
+            // try KUBECONFIG
             val kubeConfigEnv = sys.env.get("KUBECONFIG")
             kubeConfigEnv.map { kc =>
               Configuration.parseKubeconfigFile(Paths.get(kc))
-            }.getOrElse(
-              Configuration.parseKubeconfigFile()
-            ).get
+            }.getOrElse {
+              //Try to get config from a running pod -
+              // if that is not set then use default kubeconfig location
+              Configuration.useRunningPod.flatMap(_ =>
+                Configuration.parseKubeconfigFile()
```

Comment:
This will always return the results of the second operation if the first operation succeeds, which I think it not the intention.

I think it should be:

```scala
}.getOrElse {
  //Try to get config from a running pod
  Configuration.useRunningPod
}.getOrElse {
  // if that is not set then use default kubeconfig location
  Configuration.parseKubeconfigFile()
}.get
```

Reply:
True, 🤦‍♂️  well spotted, will fix